src/HealthChecks/HealthChecks/src/DependencyInjection/HealthCheckServiceCollectionExtensions.cs

```diff
@@ -12,6 +13,44 @@ namespace Microsoft.Extensions.DependencyInjection
     /// </summary>
     public static class HealthCheckServiceCollectionExtensions
     {
+        /// <summary>
+        /// Adds the <see cref="HealthCheckService"/> to the container, using the provided delegate to register
+        /// health checks.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the <see cref="HealthCheckService"/> to.</param>
+        /// <param name="configureOptions">A delegate to configure the <see cref="HealthCheckServiceOptions"/>.</param>
+        /// <returns>An instance of <see cref="IHealthChecksBuilder"/> from which health checks can be registered.</returns>
+        public static IHealthChecksBuilder AddHealthChecks(this IServiceCollection services, Action<HealthCheckServiceOptions> configureOptions)
+        {
+            if (configureOptions == null)
+            {
+                throw new ArgumentNullException(nameof(configureOptions));
+            }
+
+            services.AddOptions<HealthCheckServiceOptions>().Configure(configureOptions);
+
+            return services.AddHealthChecks();
+        }
+
+        /// <summary>
+        /// Adds the <see cref="HealthCheckService"/> to the container, using the provided delegate to register
+        /// health checks.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to add the <see cref="HealthCheckService"/> to.</param>
+        /// <param name="configureOptions">A delegate to configure the <see cref="HealthCheckServiceOptions"/>.</param>
+        /// <returns>An instance of <see cref="IHealthChecksBuilder"/> from which health checks can be registered.</returns>
+        public static IHealthChecksBuilder AddHealthChecks<TService>(this IServiceCollection services, Action<HealthCheckServiceOptions, TService> configureOptions) where TService : class
```

Comment:
Why is this needed?

Reply:
This overload allows to resolve a service from IoC container to configure the options. 
Several services now have both overloads. https://github.com/dotnet/aspnetcore/commit/5a375a7c226a3fa30720035dbc0796a664ae8a9b

Reply:
This doesn't seem worth adding, the underlying options builder api supports up to 5 services, and it's an uncommon enough scenario that it doesn't seem worth adopting as a general pattern to have all AddXyz methods add this overload.  I think we should encourage people who want to configure options with services to do it via optionsbuilder

Reply:
I removed this overload.